### PR TITLE
Fix task still available as a time entry after deleting

### PR DESCRIPTION
**Pull request by LeeSingChanYau**
This PR fixes when even after the task is completed and deleted from list , can see the task name while adding time on main.
**UPDATED**
How to test:

- Logged in as admin 
- Start timer
- Go to user profile
- Go to projects tab
- Delete a task
- Stop timer 
- Add tangible time entry 
-  Select Project/Task

Make sure that the page refreshes after you click on save. This will trigger the timer component to update the data.
Check if the deleted task shows up(hopefully not).

### CODE REVIEW

**Comment by PrabhjotSembhi** [comment]
Hi please add some more instructions. sometimes it's difficult for new volunteers to navigate the app.
adding screenshots will be really great.
I need clarity on this line on how should delete my task.
"Even after my task was deleted , I can make time entry against that task on main."

**Comment by one-community** [comment]
Yes, please make it easy for people to review your PRs Alan. Here are the guidelines if you need to review them: [Instructions for Creating a Good Pull Request on GitHub](https://docs.google.com/document/d/1d-ZzDnb_OOXXUN49V-wExHpAgOFMYpe2se1SaziA7Xw/edit#bookmark=id.1qihiwkbdteo)

**Comment by LeeSingChanYau** [comment]
@PrabhjotSembhi please test now, I gave more instructions.

**Comment by tingedudu8** [comment]
Hi @LeeSingChanYau,
I testesd the PR but it still doesn't work. I can still see the deleted task after saving the change.

https://user-images.githubusercontent.com/70128933/215607822-2462d486-63de-4bbf-813b-d175ec40f727.mov

File: src/components/UserProfile/UserProfile.jsx [hunk 1]
```diff
@@ -301,6 +301,7 @@ const UserProfile = props => {
     }
     setShouldRefresh(true);
     setChanged(false);
+    window.location.reload();
```
Comment (PrabhjotSembhi) [inline src/components/UserProfile/UserProfile.jsx:304]: Tasks are still showing after deleting them. We have to update the state of the Project task dropdown too.
I think  : )

**Review by PrabhjotSembhi — COMMENTED** [review]
Deleted tasks are still showing. please watch video 

https://user-images.githubusercontent.com/82662768/215840464-19362ee1-da61-48c2-acef-d948dcd59261.mp4

**Comment by AnishPandita** [comment]
Hello @LeeSingChanYau. After testing your PR i am seeing the same outcome as others. Deleted tasks are still there when i try to add tangible time entry after deleting the task. Also when I try to save the changes there is a alert raised saying "An error occurred while trying to save this profile." Its visible in the video. I am not sure if others also faced the issue.

https://user-images.githubusercontent.com/111321945/215888456-ba5a75e5-d24f-4700-9b1d-621ece1afa24.mov

**Comment by Gukal7** [comment]
Hi @LeeSingChanYau, I tested your code, although I didn't get any errors while deleting the task like @AnishPandita is getting but still Tasks are still showing in the project task dropdown after deleting them from the Projects area.

https://user-images.githubusercontent.com/30505274/216200872-084e6b05-fec0-4b48-9f2f-75f3620e66df.mov

**Comment by Filipe-Oliv** [comment]
Hi @LeeSingChanYau 

Still showing task name when adding time in main even after task is deleted from list.

[week 02 - 02-04-23 (1).webm](https://user-images.githubusercontent.com/114119668/216793092-ea0c9013-cfe8-4178-abcb-11e0ef834bbb.webm)

**Comment by Filipe-Oliv** [comment]
Hi @LeeSingChanYau.

I tested and it's working well.

week 03 - 02-07-23 (1)
week 03 - 02-07-23 (2)
week 03 - 02-07-23 (3)

**Comment by rauleffting** [comment]
Hi @LeeSingChanYau. I tested your PR and it's working well. 


https://user-images.githubusercontent.com/29555732/217851752-4f6ff099-9795-465a-afe0-f66234c1264c.mp4

**Review by rauleffting — APPROVED** [review]
The code is perfect. Clean and simple!

**Pull request merged**